Detectors/MUON/MCH/Clustering/src/ClusterFinderOriginal.cxx

```diff
@@ -174,7 +174,7 @@ void ClusterFinderOriginal::resetPreCluster(gsl::span<const Digit>& digits)
     double dx = mSegmentation->padSizeX(padID) / 2.;
     double dy = mSegmentation->padSizeY(padID) / 2.;
     double charge = static_cast<double>(digit.getADC()) / static_cast<double>(std::numeric_limits<unsigned long>::max()) * 1024;
-    bool isSaturated = digit.getTime().time > 0;
+    bool isSaturated = false;
```

Comment:
this trick is currently for run2 data. I suggest to use the last bit of the mNofSamples data member instead and replace it with:
bool isSaturated = (digit.nofSamples() & 0x8000) > 0;

Reply:
yes, if you implement saturation flag as a highest bit, please add proper getters and setters, in the CTF creation I'll need to split them to 2 columns to keep the dynamic range of each column as short as possible. It is cleaner to do this via proper methods.

Reply:
I was proposing to change the code inside the clustering because for the moment it is specific to run2 digits and it was not clear to me from our discussions whether we will use this bit for saturation also for run3 digits or do something else. If it's meaningful also for run3 digits then better to change the Digit structure yes.

Reply:
I would leave this possibility open for Run3 data as well, as there are way to detect saturated signals.

Reply:
@pillot in any case, if some data member of digit is going to have some bits which should not go to CTF, better have a way to strip them out via getters, propagating custom bitshifts from other code but Digits to CTF creation will be too messy.